src/plugin/file/kml/ui/placemarkedit.js

```diff
@@ -95,6 +99,21 @@ plugin.file.kml.ui.PlacemarkEditCtrl = function($scope, $element, $timeout) {
       // if creating a new annotation, expand the Annotation Options section by default
       this.defaultExpandedOptionsId = 'featureAnnotation' + this['uid'];
     }
+
+    this['customFoldersEnabled'] = true;
+
+    var folders = [];
+    var rootFolder = plugin.places.PlacesManager.getInstance().getPlacesRoot();
+    os.structs.nodeutils.flattenTree(rootFolder, folders, function(node) {
+      return /** @type {plugin.file.kml.ui.KMLNode} */ (node).isFolder();
+    });
+    this['folderOptions'] = folders;
+
+    var parentFolder = this.options['parent'];
+    if (!parentFolder && this.options['node']) {
+      parentFolder = this.options['node'].getParent();
+    }
+    this['folder'] = parentFolder ? parentFolder : rootFolder;
```

Comment:
For items which are not booleans or numbers, this is more succinctly written as `this['folder'] = parentFolder || rootFolder.`

Reply:
Fixed